fcrepo-http-api/src/main/java/org/fcrepo/http/api/services/HttpRdfService.java

```diff
@@ -136,17 +136,26 @@ public Model bodyToInternalModel(final FedoraId extResourceId, final InputStream
 
         while (stmtIterator.hasNext()) {
             final Statement stmt = stmtIterator.nextStatement();
-            if (lenientHandling && stmtIsServerManaged(stmt)) {
+            if (lenientHandling && stmtIsServerManaged(stmt) &&
+                    fedoraPropsConfig.getServerManagedPropsMode().equals(STRICT)) {
                 // Remove any statement that touches a server managed property or namespace.
                 stmtIterator.remove();
             } else {
                 try {
                     checkForDisallowedRdf(stmt);
-                } catch (final ServerManagedPropertyException | ServerManagedTypeException exc) {
-                    if (!fedoraPropsConfig.getServerManagedPropsMode().equals(RELAXED)) {
+                } catch (final ServerManagedPropertyException exc) {
+                    if (fedoraPropsConfig.getServerManagedPropsMode().equals(STRICT)) {
                         exceptions.add(exc);
                         continue;
                     }
+                } catch (final ServerManagedTypeException exc) {
```

Comment:
This approach still allows me to provide non-relaxed SMTs and have them get silently ignored rather than 409'ed without specifying lenient behavior. For instance:
```
echo "<> <http://www.loc.gov/premis/rdf/v1#hasMessageDigest> 'helloworld'" |
curl -ufedoraAdmin:fedoraAdmin -XPUT http://localhost:8080/rest/binary_with_digests/fcr:metadata --data-binary @- -H"Content-Type: text/turtle" -H"Prefer: handling=lenient; received='minimal'"
```
Gives a 204, but ignores the digest property. 

What would you think about changing `ServerManagedTypeException` to be a subclass of `ServerManagedPropertyException`, and then introducing another subclass named `RelaxedServerManagedPropertyException` which is only thrown when relaxed properties are encountered? Then for blocks like this you could just handle `RelaxedServerManagedPropertyException` vs `ServerManagedPropertyException`, since my understanding is that all other SMTs should be treated the same in terms of when to throw exceptions or not.

Reply:
Ok I see what you are saying, I'll add a new exception and split out the tests.

Reply:
Also, in your example you **did** specify lenient behaviour.

Reply:
Sorry, I ran it both ways to verify the behavior, the correct command is
```
echo "<> <http://www.loc.gov/premis/rdf/v1#hasMessageDigest> 'helloworld'" |
curl -ufedoraAdmin:fedoraAdmin -XPUT http://localhost:8080/rest/binary_with_digests/fcr:metadata --data-binary @- -H"Content-Type: text/turtle"
```
which gives the same 204 response